libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -453,17 +459,20 @@ bool NavEKF3_core::readyToUseRangeBeacon(void) const
 // return true if we should use the compass
 bool NavEKF3_core::use_compass(void) const
 {
-    return effective_magCal() != MagCal::EXTERNAL_YAW && _ahrs->get_compass() && _ahrs->get_compass()->use_for_yaw(magSelectIndex) && !allMagSensorsFailed;
+    return effectiveMagCal != MagCal::EXTERNAL_YAW &&
+           effectiveMagCal != MagCal::GSF_YAW &&
+           _ahrs->get_compass() &&
+           _ahrs->get_compass()->use_for_yaw(magSelectIndex) &&
+           !allMagSensorsFailed;
 }
 
-// are we using an external yaw source? Needed for ahrs attitudes_consistent
+// are we using a yaw source other than the magnetomer?
 bool NavEKF3_core::using_external_yaw(void) const
 {
-    MagCal mag_cal = effective_magCal();
-    if (mag_cal != MagCal::EXTERNAL_YAW && mag_cal != MagCal::EXTERNAL_YAW_FALLBACK) {
-        return false;
+    if (effectiveMagCal == MagCal::EXTERNAL_YAW || effectiveMagCal == MagCal::EXTERNAL_YAW_FALLBACK || effectiveMagCal == MagCal::GSF_YAW) {
+        return AP_HAL::millis() - last_gps_yaw_fusion_ms < 5000 || AP_HAL::millis() - lastSynthYawTime_ms < 5000;
```

Comment:
should only call AP_HAL::millis() once, using a local var

Reply:
I'll make the lastSynthYawTime_ms and last_gps_yaw_fusion_ms timers  use the fusion time horizon time step imuSampleTime_ms as the reference